packages/keyring-controller/src/KeyringController.ts

```diff
@@ -2277,6 +2277,9 @@ export class KeyringController extends BaseController<
    */
   #updateVault(): Promise<boolean> {
     return this.#withVaultLock(async () => {
+      // Ensure no duplicate accounts are persisted.
+      await this.#assertNoDuplicateAccounts();
```

Comment:
An eventual error thrown here will trigger a rollback in all public functions that mutate the vault. That is, any new account added before this line and after the last vault update will be discarded

Reply:
Also, just quoting you here: https://github.com/MetaMask/core/pull/5710#issuecomment-2828943571

> (the vault is not updated during login, only decrypted to restore keyrings).

Thus, even if a user already have duplicated accounts in its state (prior to that change), this won't brick his wallet 👍